pkg/docker/storage.jsx

```diff
@@ -0,0 +1,511 @@
+        init_model(function (model) {
+            dfd.resolve(model);
+        });
+
+        model_promise = dfd.promise();
+        return model_promise;
+    }
+
+    var DriveBox = React.createClass({
+        getInitialState: function () {
+            return {
+                drives: this.props.model.additional_drives,
+                checked: { }
+            };
+        },
+        componentDidMount: function () {
+            var self = this;
+            $(this.props.model).on("changed", function () {
+                self.setState({ drives: self.props.model.additional_drives });
+            });
+        },
+
+        driveChecked: function (drive) {
+            return !!this.state.checked[drive.dev];
+        },
+        makeDriveToggler: function (drive) {
+            var self = this;
+            return function (event) {
+                self.state.checked[drive.dev] = !self.state.checked[drive.dev];
+                $(event.currentTarget).find("input").prop("checked", self.state.checked[drive.dev]);
+                self.setState({ checked: self.state.checked });
+            };
+        },
+
+        onButtonClicked: function () {
+            var self = this;
+            if (self.props.callback) {
+                var drives = [ ];
+                var devs = [ ];
+                for (var d in self.state.checked) {
+                    if (self.state.checked[d]) {
+                        for (var dd in self.state.drives) {
+                            if (self.state.drives[dd].dev == d) {
+                                drives.push(self.state.drives[dd].drive);
+                                devs.push(self.state.drives[dd].dev);
+                            }
+                        }
+                    }
+                }
+                self.setState({ checked: { } });
+                self.props.callback(drives, devs);
+            }
+        },
+
+        render: function() {
+            var self = this;
+            var i, d;
+
+            var button_enabled = false;
+            var drive_rows = [ ];
+            var drive_devs = { };
+
+            for (i = 0; i < self.state.drives.length; i++) {
+                var drive = self.state.drives[i];
+
+                if (self.state.checked[drive.dev])
+                    button_enabled = true;
+
+                drive_devs[drive.dev] = true;
+
+                drive_rows.push(
+                    <tr onClick={self.makeDriveToggler(drive)}>
+                        <td><input type="checkbox"
+                                   checked={self.driveChecked(drive)}/>
+                        </td>
+                        <td><img src="images/drive-harddisk-symbolic.svg"/></td>
+                        <td><span>{drive.name}</span><br/><span>{drive.desc}</span></td>
+                    </tr>);
+            }
+
+            for (d in self.state.checked) {
+                if (!drive_devs[d])
+                    self.state.checked[d] = false;
```

Comment:
It matters in this scenario:
- check a disk
- hit "add" and confirm the dialog
- the disk disappears from the list and appears in the pool
- hit "reset" and confirm the dialog
- the disk disappears from the pool and appears again in this list

If we don't reset the "checked" state for disks that are not currently in the list, they will be checked when they appear again.